### ISSUE TITLE
support for user authentication without access to master db

### ISSUE THREAD

**Issue by mabster** [Azure/sql-action#89]
Hi gang,

I'm trying to use Azure/sql-action in the "deploy" phase of my workflow when deploying an ASP.NET Core app to Azure.

My Azure SQL server definitely has "Allow Azure services and resources to access this server" set to "Yes", but when my workflow runs I get the following error:

Error: Failed to add firewall rule. Unable to detect client IP Address. Sqlcmd: Error: Microsoft ODBC Driver 17 for SQL Server : Login failed for user '(redacted)'..

My code looks like this:

   - name: Apply EF migration script
        uses: Azure/sql-action@v1
        with:
          server-name: (my server name).database.windows.net
          connection-string: ${{ secrets.CONNECTIONSTRING_PUCKA }}
          sql-file: ./migrate.sql

In the server-name attribute I've tried it with and without the "tcp:" prefix (which is in the connection string) but I get the same error both ways.

Your README says if I have "allow azure services" set to "yes" then I should be fine. Am I doing something wrong?

**Comment by mabster** [comment]
Hmm I redacted the username and server name in my post above but I used angle brackets so GitHub removed the word "redacted" entirely. Rest assured the user id is set in the connection string. :)

**Comment by mabster** [comment]
Have updated to the latest version but still getting the same error. My action looks like this now:

```
 - name: Apply EF migration script
        uses: Azure/sql-action@v1.2
        with:
          connection-string: ${{ secrets.CONNECTIONSTRING_PUCKA }}
          sql-file: ./migrate.sql
```

**Comment by mabster** [comment]
OK I have a theory.

When I created my Azure SQL user, I did it in my database like this:

```
CREATE USER [my-user] WITH PASSWORD = '...';
ALTER ROLE db_owner ADD MEMBER [my-user];
```

So the user exists in my database but does not correspond to a "login". It therefore has no access to the master db.

When you try to detect the client IP address, you're doing this:

```
await exec.exec(`"${sqlCmdPath}" -S ${serverName} -U "${connectionString.userId}" -P "${connectionString.password}" -Q "select getdate()"`
```

... which is trying to connect to my server, but not to my database. So I'm guessing it will try to connect to master and fail.

Can you use the full connection string in your command so that it connects to the database directly rather than to master?

**Comment by mabster** [comment]
Or alternatively, could this line:

https://github.com/Azure/sql-action/blob/357d609def12a80356de4df4aeec7d87cf9aed77/src/SqlUtils.ts#L15

... have a -d "${connectionString.database}" parameter? My connection string says "Initial Catalog=..." but presumably "database' is the right property name for your connectionString object.

**Comment by mabster** [comment]
Have submitted a PR for this! Hope it's OK!

**Comment by zijchen** [comment]
This was fixed in #104
